distributed/scheduler.py

```diff
@@ -4772,9 +4760,12 @@ def handle_worker_status_change(
                 worker_msgs: dict = {}
                 self._transitions(recs, client_msgs, worker_msgs, stimulus_id)
                 self.send_all(client_msgs, worker_msgs)
-
-        else:
-            self.running.discard(ws)
+        elif ws.status == Status.paused:
+            self.running.remove(ws)
+        elif ws.status == Status.closing:
+            await self.remove_worker(
```

Comment:
⚠️ This feels like a substantive change that's built around the partially-removed state: https://github.com/dask/distributed/issues/6390. What's the motivation for this? Unless this is strictly necessary, I'd be more comfortable leaving it out, since we know there are weird race conditions around `remove_worker(..., close=False)` https://github.com/dask/distributed/issues/6354, though they're lessened by having removed reconnect.

Reply:
FWIW once we're in this state, the worker is already in `Worker.close`. Worker.close will eventually close the stream once it is done

I am very certain that I added this code path to ensure tests pass. I believe it was connected to a test in deploy but I do not recall what the problem was exactly.
I remember it being connected to removing the worker from `self.running` w/out properly removing everything else.